src/main/proto/ga4gh/genotype_phenotype_service.proto

```diff
@@ -0,0 +1,186 @@
+   Phenotype:
+   Human Phenotype Ontology (HPO): http://www.ontobee.org/browser/index.php?o=hp
+   Disease Ontology (DO): http://purl.obolibrary.org/obo/DOID_4
+   Feature:
+   Sequence Ontology (SO): http://www.sequenceontology.org/browser/
+   Evidence:
+   Evidence Code Ontology (ECO): http://www.ontobee.org/browser/index.php?o=ECO
+   Ontology for Biomedical Investigations (OBI): http://purl.obofoundry.org/obo/obi/browse
+*/
+
+service GenotypePhenotypeService {
+
+    rpc SearchPhenotypeAssociationSets(SearchPhenotypeAssociationSetsRequest)
+        returns (SearchPhenotypeAssociationSetsResponse);
+
+    rpc SearchPhenotype(SearchPhenotypesRequest)
+        returns (SearchPhenotypesResponse);
+
+    // see SequenceAnnotations.SearchFeatureSets
+
+    rpc GetPhenotypeAssociations(SearchGenotypePhenotypeRequest)
+        returns (SearchGenotypePhenotypeResponse);
+
+}
+
+/******************  /phenotypeassociationsets/search  *********************/
+// This request maps to the body of `POST /phenotypeassociationsets/search` as JSON.
+message SearchPhenotypeAssociationSetsRequest {
+    // The `Dataset` to search. Mandatory
+    string dataset_id = 1;
+
+    // Specifies the maximum number of results to return in a single page.
+    // If unspecified, a system default will be used.
+    int64 page_size = 2;
+
+    // The continuation token, which is used to page through large result sets.
+    // To get the next page of results, set this parameter to the value of
+    // `nextPageToken` from the previous response.
+    string page_token = 3;
+}
+// This is the response from `POST /phenotypeassociationsets/search` expressed as JSON.
+message SearchPhenotypeAssociationSetsResponse {
+    // The list of matching phenotype association sets.
+    repeated PhenotypeAssociationSet phenotype_association_sets = 1;
+
+    // The continuation token, which is used to page through large result sets.
+    // Provide this value in a subsequent request to return the next page of
+    // results. This field will be empty if there aren't any additional results.
+    string next_page_token = 2;
+}
+
+/******************  /phenotypes/search  *********************/
+
+// One or more ontology terms can be queried together.
+message OntologyTermQuery {
+    repeated OntologyTerm terms = 1;
+}
+
+// One or more ids can be queried together.  Generally used for instances
+// of a particular class of object (e.g. a specific gene or SNP).
+message ExternalIdentifierQuery {
+    repeated ExternalIdentifier ids = 1;
+}
+
+// Evidence for the phenotype association.
+message EvidenceQuery {
+    // ECO or OBI is recommended
+    OntologyTerm evidenceType = 1;
+
+    string description = 2;  /*regex*/
+
+    repeated ExternalIdentifier external_identifiers = 3; /* new field */
+}
+
+message SearchPhenotypesRequest {
+    // The `PhenotypeAssociationSet` to search. Mandatory
+    string phenotype_association_set_id = 1;
+
+    string id = 2;
```

Comment:
Why is the `id` included in the search request? If you have an ID, would you be searching for it?